clusterloader2/framework/framework.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/framework/util"
+)
+
+const (
+	automanagedNamespaceName = "namespace"
+	contentType              = "application/vnd.kubernetes.protobuf"
+	qps                      = 100
+	burst                    = 200
+)
+
+// Framework allows for interacting with Kubernetes cluster via
+// official Kubernetes client.
+type Framework struct {
+	// TODO(krzysied): Change this to a set of FrameworkClients.
+	clientSet     clientset.Interface
+	dynamicClient dynamic.Interface
+}
+
+// CreateAutomanagedNamespaces creates automanged namespaces.
+func (f *Framework) CreateAutomanagedNamespaces(namespaceCount int) error {
+	for i := 0; i < namespaceCount; i++ {
+		if err := util.CreateNamespace(f.clientSet, fmt.Sprintf("%v-%d", automanagedNamespaceName, i)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteAutomanagedNamespaces deletes automanged namespaces.
```

Comment:
all automanaged

Reply:
I'm removing namespaces count parameter. Automaneged namespaces count will be part of framework state.
Delete will remove all of automanaged namespaces. Addition Create is assumed to not be used without Delete called first.